### PR TITLE
Merge Bump Hunter Preliminary Analysis Code into Master

### PR DESCRIPTION
**Pull request by mccaky**
The bump hunter code has been significantly advanced to the point of a preliminary analysis code. It should be merged into the master.

### CODE REVIEW

File: analysis/include/BumpHunter.h [hunk 1]
```diff
@@ -36,21 +36,20 @@
 
 //---//
 #include <HpsFitResult.h>
```
Comment (mholtrop) [inline analysis/include/BumpHunter.h]: Use "" instead of <> for local include files.

File: analysis/include/BumpHunter.h [hunk 2]
```diff
@@ -36,21 +36,20 @@
...
+#include <ChebyshevFitFunction.h>
```
Comment (mholtrop) [inline analysis/include/BumpHunter.h]: Use "" instead of <> for local include files.

File: analysis/include/ChebyshevFitFunction.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#ifndef __CHEBYSHEV_FUNC_H__
+#define __CHEBYSHEV_FUNC_H__
+
+#include <FitFunction.h>
```
Comment (mholtrop) [inline analysis/include/ChebyshevFitFunction.h]: Use "" instead of <> for local include files.

File: analysis/include/FitFunction.h [hunk 4]
```diff
@@ -0,0 +1,121 @@
+#ifndef FITFUNCTION_H
+#define FITFUNCTION_H
+#include <TMath.h>
+#include <FunctionMath.h>
```
Comment (mholtrop) [inline analysis/include/FitFunction.h]: Use "" instead of <> for local include files.

File: analysis/include/HpsFitResult.h [hunk 5]
```diff
@@ -10,6 +10,8 @@
 #include <TFitResult.h>
 #include <TFitResultPtr.h>
 
+#include <ChebyshevFitFunction.h>
```
Comment (mholtrop) [inline analysis/include/HpsFitResult.h]: Use "" instead of <> for local include files.

File: analysis/src/BumpHunter.cxx [hunk 6]
```diff
@@ -11,52 +11,51 @@
 
 #include <BumpHunter.h>
```
Comment (mholtrop) [inline analysis/src/BumpHunter.cxx]: Use "" instead of <> for local include files.

File: analysis/src/ChebyshevFitFunction.cxx [hunk 7]
```diff
@@ -0,0 +1,7 @@
+#include <ChebyshevFitFunction.h>
+#include <FunctionMath.h>
```
Comment (mholtrop) [inline analysis/src/ChebyshevFitFunction.cxx]: Use "" instead of <> for local include files.

File: analysis/src/FunctionMath.cxx [hunk 8]
```diff
@@ -0,0 +1,44 @@
+#include <FunctionMath.h>
```
Comment (mholtrop) [inline analysis/src/FunctionMath.cxx]: Use "" instead of <> for local include files.

**Comment by cbravo135** [comment]
Yeah, what Maurik is saying is definitely the preferred way to include headers from the local project. <> tends to be reserved for standard library includes. Please push a commit with this change.

**Comment by mccaky** [comment]
I changed the brackets to quotes for the local includes. I left the ROOT includes in brackets.

**Comment by pbutti** [comment]
Hi All,

Can we also add some sort of instructions / documentation about the code that is going to be pushed with this PR? Would be good to have examples in the readme on how the code is used for further reference.

**Comment by pbutti** [comment]
For the rest, I'm OK with merging

**Review by pbutti — APPROVED** [review]

**Pull request merged**